media-proxy/src/mesh/st2110.cc

```diff
@@ -130,13 +130,25 @@ void get_mtl_dev_params(mtl_init_params& st_param, const std::string& dev_port,
     if (getenv("KAHAWAI_CFG_PATH") == NULL) {
         setenv("KAHAWAI_CFG_PATH", "/usr/local/etc/imtl.json", 0);
     }
+    if (getenv("MEDIA_PROXY_MAIN_LCORE") != NULL) {
+        try {
+            st_param.main_lcore = std::stoul(std::string(getenv("MEDIA_PROXY_MAIN_LCORE")));
+        } catch (const std::invalid_argument& e) {
+            log::error("Conversion failed, Env MEDIA_PROXY_MAIN_LCORE must be an integer");
+            st_param.main_lcore = 0;
+        } catch (const std::out_of_range& e) {
+            log::error("Conversion failed, Env MEDIA_PROXY_MAIN_LCORE is out of range");
+            st_param.main_lcore = 0;
+        }
+    }
     strlcpy(st_param.port[MTL_PORT_P], dev_port.c_str(), MTL_PORT_MAX_LEN);
     inet_pton(AF_INET, ip_addr.c_str(), st_param.sip_addr[MTL_PORT_P]);
     st_param.pmd[MTL_PORT_P] = mtl_pmd_by_port_name(st_param.port[MTL_PORT_P]);
     st_param.num_ports = 1;
     st_param.flags = MTL_FLAG_BIND_NUMA;
     st_param.flags |= MTL_FLAG_TX_VIDEO_MIGRATE;
     st_param.flags |= MTL_FLAG_RX_VIDEO_MIGRATE;
+    st_param.flags |= MTL_FLAG_RX_SEPARATE_VIDEO_LCORE;
```

Comment:
I am not MTL expert, but should this flag be set only if st_param.main_lcore and/or st_param.lcores is set ?

Reply:
This is just for separation of lcores (threads) for input and output. If the lcores are not provided, MtlManager or mtl itself will assign the core that is available.